File: solutions/2439. Minimize Maximum of Array/2439.py
class Solution:
  def minimizeArrayValue(self, nums: list[int]) -> int:
    ans = 0
    prefix = 0

    for i, num in enumerate(nums):
      prefix += num
      prefixAvg = math.ceil(prefix / (i + 1))
      ans = max(ans, prefixAvg)

    return ans

File: solutions/2439. Minimize Maximum of Array/2439.java
class Solution {
  public int minimizeArrayValue(int[] nums) {
    long ans = 0;
    long prefix = 0;

    for (int i = 0; i < nums.length; ++i) {
      prefix += nums[i];
      final long prefixAvg = (long) Math.ceil(prefix / (double) (i + 1));
      ans = Math.max(ans, prefixAvg);
    }

    return (int) ans;
  }
}
